server/database.sql

```diff
@@ -1,150 +1,378 @@
+/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
+/*!40103 SET TIME_ZONE='+00:00' */;
+/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
+/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
+/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
+/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
+
+--
+-- Table structure for table `contests`
+--
+
+DROP TABLE IF EXISTS `contests`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `show_leaderboard` tinyint NOT NULL DEFAULT '0',
+  `public` tinyint NOT NULL DEFAULT '1',
+  `creator` varchar(45) NOT NULL,
+  `name` text NOT NULL,
+  `start_time` timestamp NOT NULL,
+  `end_time` timestamp NOT NULL,
+  `about` text,
+  `rules` text,
+  `prizes` text,
+  `confidential_questions` tinyint DEFAULT '0',
+  `participants_count` int DEFAULT '0',
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_users_idx` (`creator`),
+  CONSTRAINT `fk_contests_users` FOREIGN KEY (`creator`) REFERENCES `users` (`username`) ON DELETE RESTRICT ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `contests_groups`
+--
+
+DROP TABLE IF EXISTS `contests_groups`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests_groups` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `contest_id` int NOT NULL,
+  `group_id` int NOT NULL,
+  PRIMARY KEY (`id`),
+  KEY `fk_contests_groups1_idx` (`contest_id`),
+  KEY `fk_contests_groups_2_idx` (`group_id`),
+  CONSTRAINT `fk_contests_groups_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_groups_2` FOREIGN KEY (`group_id`) REFERENCES `groups` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `contests_moderators`
+--
+
+DROP TABLE IF EXISTS `contests_moderators`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests_moderators` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `contest_id` int NOT NULL,
+  `moderator` varchar(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_moderators_1_idx` (`contest_id`),
+  KEY `fk_contests_moderators_2_idx` (`moderator`),
+  CONSTRAINT `fk_contests_moderators_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_moderators_2` FOREIGN KEY (`moderator`) REFERENCES `users` (`username`) ON DELETE RESTRICT ON UPDATE CASCADE
```

Comment:
Similarly, 1 more constraints needs to be added to avoid unnecessary duplicate entries -
```
UNIQUE KEY `unique_contests_moderators` (`contest_id`,`moderator`)
```